uberenv.py

```diff
@@ -361,16 +389,30 @@ def setup_paths_and_dirs(self):
             if not uberenv_plat is None:
                 self.spack_config_dir = pabs(pjoin(spack_configs_path,uberenv_plat))
 
-        # Spack local packages path
-        spack_packages_path = pabs(pjoin(self.uberenv_path,"packages"))
+        # Find project level packages to override spack's internal packages
         if "spack_packages_path" in self.project_opts.keys():
-            new_path = self.project_opts["spack_packages_path"]
-            if new_path is not None:
-                spack_packages_path = pabs(new_path)
-        # All local packages
-        self.pkgs = pjoin(spack_packages_path, "*")
+            # packages directories listed in project.json
+            _paths = self.project_opts["spack_packages_path"]
+            if type(_paths) is str:
+                # user gave a single string
+                self.append_path_to_packages_paths(_paths)
+            else:
+                # user gave a list of strings
+                for _path in _paths:
+                    self.append_path_to_packages_paths(_path)
+        else:
+            # default to packages living next to uberenv script if it exists
+            self.append_path_to_packages_paths(pjoin(self.uberenv_path,"packages"), errorOnNonexistant=False)
```

Comment:
I made this default path optional in case someone wanted to not provide their own packages